File: Projects/Project_03_1D_Burger/For_Students/julia/exact_solution.jl
#---------------------------------------------------------------------#
#This function computes the Initial & Exact Solutions.
#Written by F.X. Giraldo on April 19, 2019
#           Department of Applied Mathematics
#           Naval Postgraduate School
#           Monterey; CA 93943-5216
#---------------------------------------------------------------------#
function exact_solution(coord,Npoin,time,case,DFloat)

    #Set some constants
    xmin=0
    xmax=+2
    xc=0.5*(xmax+xmin)
    xl=xmax-xmin
    rc=0.125
    sigma=1/16; #best for right convergence rates
    # sigma=1/8
    # sigma=1/4; #fatter Gaussian to compare against 2D code
    
    #Initialize
    qe=zeros(DFloat,Npoin)

    #timec=time - floor(time)
    timec=time
    #Generate Grid Points
    for I=1:Npoin
        x=coord[I]
        if (case == 1) #Sin
            qe[I]=sin(pi*x) + 0.01;
        end
    end #I
    return (qe)
end

File: Projects/Project_02b_1D_Wave_Laguerre/For_Instructors/Old_1D_Advection_Laguerre_V2/exact_solution.m
%---------------------------------------------------------------------%
%This function computes the Initial and Exact Solutions.
%Written by F.X. Giraldo on 10/2003
%           Department of Applied Mathematics
%           Naval Postgraduate School 
%           Monterey, CA 93943-5216
%---------------------------------------------------------------------%
function [qe,u0] = exact_solution(intma,coord,npoin,nelem,nelem_LGL,ngl)

%Initialize
qe=zeros(npoin,1);

%Set some constants
xmin=coord(1,1);
xmax=coord(ngl(nelem_LGL),nelem_LGL);
%xm=0.5*(xmax + xmin);
xm=0.5*(xmax + xmin);
xc=xm;
u0=2;

%Generate Solution
for e=1:nelem
    for i=1:ngl(e)
        x=coord(i,e);
        ip=intma(i,e);
        %qe(ip)=exp(-0.1*(x-xc)^2);
        %qe(ip)=exp(-100*(x-xc)^2);
        qe(ip)=exp(-16*(x-xc)^2);
    end %i
end %e  


      
